### PR TITLE
Update runtimeconfig.json doc for 2.1

### PR DESCRIPTION
**Pull request by steveharter**
Fixes https://github.com/dotnet/cli/issues/7861.

Update terminology, samples, and add new information for 2.1.0. There is still some detail regarding project.lock.json and other items that I didn't modify.

### CODE REVIEW

File: Documentation/specs/runtime-configuration-file.md [hunk 1]
```diff
@@ -2,206 +2,155 @@
...
+* `MyApp.runtimeconfig.json` - An **optional** configuration file containing runtime configuration settings. This file is required for framework-dependent applications, but not for self-contained apps.
+* `MyApp.runtimeconfig.dev.json` - An **optional** configuration file containing runtime configuration settings that typically only exists in a non-published output and thus is used for development-time scenarios. This file typically specifies additional probing paths. Depending on the semantics of each setting, the setting is either combined with or overridden by the values from `MyApp.runtimeconfig.json`.
+* `MyApp.deps.json` - A list of dependencies, as well as compilation context data, compilation dependencies and version information used to address assembly conflicts. Not technically required, but required to use the servicing or package cache/shared package install features. If the file is not present, all assemblies in the current folder are used instead.
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: I think `compilation dependencies` is redundant with `compilation context data`, isn't it?

File: Documentation/specs/runtime-configuration-file.md [hunk 2]
```diff
@@ -2,206 +2,155 @@
...
+* `applyPatches` - When `false`, the framework version is strictly obeyed by the host. When `applyPatches` is unspecified or `true`, the framework from either the same or a higher version that differs only by the patch field will be used.
   * For example, if `version=1.0.1` and `applyPatches` is `true`, the host would load the shared framework from `1.0.{n}`, where `n >= 1`, but will not load from `1.1.0`, even if present. When `applyPatches` is `false`, the shared framework will be loaded from `1.0.1` strictly.
-  * **Note:** This does not apply to `SemVer`'s `prerelease` versions, but only for `production` releases.
-  * **Note:** This section will not be used for standalone applications that do not rely upon a shared framework.
-
-* Others _TBD_
+  * **Note:** This does not apply to `pre-release` versions; it applies only to `production` releases.
+  * **Note:** This section should not exist self-contained applications because they do not rely upon a shared framework.
+* `rollForwardOnNoCandidateFx` - Determines roll-forward behavior of major and minor. Only applies to `production` releases. Values: 0(Off), 1 (roll forward on [minor]), Roll forward on [major] and [minor]
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: What's the value for `Roll forward on [major] and [minor]`?  We have `0`, `1` for the other two, but this one doesn't seem to have a value.

File: Documentation/specs/runtime-configuration-file.md [hunk 3]
```diff
@@ -2,206 +2,155 @@
...
+ See [roll-forward-on-no-candidate documentation](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/roll-forward-on-no-candidate-fx.md) for more information.
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: Can you use a relative link here?  They are easier to maintain across branches.

File: Documentation/specs/runtime-configuration-file.md [hunk 4]
```diff
@@ -2,206 +2,155 @@
...
 
-These settings are read by `corehost` to determine how to initialize the runtime. All versions of `corehost` **must ignore** settings in this section that they do not understand (thus allowing new settings to be added in later versions).
+These settings are read by `apphost` to determine how to initialize the runtime. All versions of `apphost` **must ignore** settings in this section that they do not understand (thus allowing new settings to be added in later versions).
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: IMO we shouldn't use "apphost" to mean both the `apphost.exe` and `dotnet.exe`.  Can we just say "host" when we mean them both?

File: Documentation/specs/runtime-configuration-file.md [hunk 5]
```diff
@@ -2,206 +2,155 @@
...
-
-```json
-{
-    "compilationOptions": {
-        "allowUnsafe": true,
-        "defines": [ "CORE_CLR", "DEBUG_MODE" ]
-    }
-}
-```
+This section is created during build from the project's settings. The exact settings found here are specific to the compiler that produced the original application binary. Some example settings include: `defines`, `languageVersion` (C#/VB), `allowUnsafe` (C#), etc.
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: "`languageVersion` (C#/VB)" makes it sound like C# or VB is the possible values for these setting, when in reality it is a number like `7.3`

File: Documentation/specs/runtime-configuration-file.md [hunk 6]
```diff
@@ -338,9 +236,9 @@ The shared framework's deps file will also contain a `runtimes` section defining
 
 The host will have a RID embedded in it during compilation (for example, `win10-x64` for Windows 64-bit). It will look up the corresponding entry in the `runtimes` section to identify what the fallback list is for `win10-x64`. The fallbacks are identified from most-specific to least-specific. In the case of `win10-x64` and the example above, the fallback list is: `"win10-x64", "win10", "win81-x64", "win81", "win8-x64", "win8", "win7-x64", "win7", "win-x64", "win", "any", "base"` (note that an exact match on the RID itself is the first preference, followed by the first item in the fallback list, then the next item, and so on).
```
Comment (eerhardt) [inline Documentation/specs/runtime-configuration-file.md]: `The host will have a RID embedded in it during compilation`
This line is incorrect. Can you fix it?  We do RID detection in the host now.

File: Documentation/specs/runtime-configuration-file.md [hunk 7]
```diff
@@ -2,206 +2,155 @@
 
 The runtime configuration files store the dependencies of an application (formerly stored in the `.deps` file). They also include runtime configuration options, such as the Garbage Collector mode. Optionally they can also include data for runtime compilation (compilation settings used to compile the original application, and reference assemblies used by the application).
 
-**Note:** This document doesn't provide full explanations as to why individual items are needed in this file. That is covered in the [`corehost` spec](corehost.md) and via the `Microsoft.Extensions.DependencyModel` assembly.
+**Note:** This document doesn't provide full explanations as to why individual items are needed in this file. That is covered in the [`apphost` spec](corehost.md) and via the `Microsoft.Extensions.DependencyModel` assembly.
```
Comment (nguerrera) [inline Documentation/specs/runtime-configuration-file.md]: Aside: is this the best place for this document? Should it go in core-setup? Or docs.microsoft.com?

File: Documentation/specs/runtime-configuration-file.md (hunk 3 above)
Comment (steveharter) [reply Documentation/specs/runtime-configuration-file.md]: I don't think relative across branches works. Also, if someone has a local branch the link probably won't work.

File: Documentation/specs/runtime-configuration-file.md (hunk 4 above)
Comment (steveharter) [reply Documentation/specs/runtime-configuration-file.md]: Done

File: Documentation/specs/runtime-configuration-file.md (hunk 5 above)
Comment (steveharter) [reply Documentation/specs/runtime-configuration-file.md]: Changing to "(e.g. the version of C# or VB)"

File: Documentation/specs/runtime-configuration-file.md (hunk 6 above)
Comment (steveharter) [reply Documentation/specs/runtime-configuration-file.md]: Chaning to "The host will detect the RID at runtime"

File: Documentation/specs/runtime-configuration-file.md (hunk 7 above)
Comment (steveharter) [reply Documentation/specs/runtime-configuration-file.md]: Yes I could see the "corehost" spec be moved to core-setup documentation to be consistent with other related documents also located there.

**Comment by steveharter** [comment]
Any other comments? I need at least one "approval" to commit. Thanks

**Pull request merged**